### PR TITLE
Feature/crud uni tot components

### PR DESCRIPTION
**Pull request by LiWeiYeh**

### CODE REVIEW

File: src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
+            <input type="text" class="min-w-10rem w-full" pInputText [(ngModel)]="property"
+                [required]="exprIsTot ? true : false" />
+        </div>
     </div>
-$else$
-    <span><i>not readable</i></span>
-$endif$ -->
+</div>
+
+<ng-template #exprIsNotUni>
```
Comment (LiWeiYeh) [inline src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html:10]: Is it possible to make these templates generic?

File: src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
+
+<ng-template #exprIsNotUni>
+    <div class="text-center w-full">
+        <div *ngFor="let row of requireArray(property); index as i">
+            <span my-showonhover-box class="glyphicon glyphicon-minus" uib-tooltip="Remove" tooltip-trigger="mouseenter"
+                tooltip-placement="left"></span>
+            <span class="wordwrap">{{row}}</span>
+        </div>
+        <div>
+            <!-- <textarea ng-model="selected.value" placeholder="Add text" class="form-control input-sm" rows="6" cols="80" my-change-on-blur="addItem(resource, '$name$', selected, patchResource)"$if(exprIsTot)$ ng-required="resource['$name$'].length === 0"$endif$></textarea> -->
```
Comment (LiWeiYeh) [inline src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html:18]: Is this comment still needed? If not, delete.

**Comment by MarkCorazon** [comment]
On a second thought; I think this pull request is made too early as it doesn't implement the create & delete functionality (It should at least show a plus and a trashcan icon.). The injective logic is also not fully implemented as its logic is still commented.

**Comment by MarkCorazon** [comment]
Still the create & delete functionality are not implemented as it is also not part of the component code of the AmpersandTarski/prototype repo. The injective logic is now implemented as it shows the plus and minus icons.

File: src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html (hunk 1 above)
Comment (MarkCorazon) [reply src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html:10]: This should be possible, and is for sure a good idea. Although I'm not sure how to manage this to be accessable for every component.

File: src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html (hunk 2 above)
Comment (MarkCorazon) [reply src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html]: I updated this to work on the injective case. PR is reopened.

File: src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
-                <div class="wordwrap">{{resource['$name$']}}</div>
-            $else$
-                <div ng-repeat="row in resource['$name$'] track by \$index" class="wordwrap">{{row}}</div>
-            $endif$
-        $endif$
+<div *ngIf="canRead; else notReadable">
+    <div class="p-inputgroup" *ngIf="canUpdate; else notUpdatable">
+        <div class="w-full" *ngIf="exprIsUni; else exprIsNotUni">
+            <input type="text" class="min-w-10rem w-full" pInputText [(ngModel)]="property"
+                [required]="exprIsTot ? true : false" />
```
Comment (LiWeiYeh) [inline src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html:5]: you can leave it like `[required]="exprIsTot ? true : false"`, the true and false is redundant

File: src/app/shared/atomic-components/atomic-boolean/atomic-boolean.component.html [hunk 4]
```diff
@@ -1,2 +1,16 @@
-<p-inputSwitch *ngIf="exprIsUni && canRead" [(ngModel)]="property" [disabled]="canUpdate ? false : true"
-    (click)="getState()"></p-inputSwitch>
\ No newline at end of file
+<div *ngIf="canRead; else notReadable">
+    <div class="p-inputgroup" *ngIf="exprIsUni; else exprIsNotUni">
+        <div class="w-full">
+            <p-inputSwitch *ngIf="exprIsUni && canRead" [(ngModel)]="property" [disabled]="canUpdate ? false : true"
```
Comment (LiWeiYeh) [inline src/app/shared/atomic-components/atomic-boolean/atomic-boolean.component.html:4]: `[disabled]="canUpdate ? false : true"` can be changed to `disabled]="!canUpdate`. The `!`-operator inverses the true to false and vice versa

File: src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html (hunk 3 above)
Comment (MarkCorazon) [reply src/app/shared/atomic-components/atomic-alphanumeric/atomic-alphanumeric.component.html:5]: Great addition! Fully overlooked.

File: src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html [hunk 5]
```diff
@@ -1,28 +1,38 @@
-<textarea *ngIf="canRead && canUpdate && exprIsUni" pInputTextarea [(ngModel)]="property"
-    [required]="exprIsTot ? true : false" maxlength="64000"></textarea>
+<div *ngIf="canRead; else notReadable">
+    <div class="p-inputgroup" *ngIf="canUpdate; else notUpdatable">
+        <div class="w-full" *ngIf="exprIsUni; else exprIsNotUni">
+            <textarea pInputTextarea [(ngModel)]="property" [required]="exprIsTot ? true : false" maxlength="64000"
```
[the remark this answers is not in the document]
Comment (MarkCorazon) [reply src/app/shared/atomic-components/atomic-bigalphanumeric/atomic-bigalphanumeric.component.html]: Great addition! Fully overlooked

**Pull request merged**